internal/mode/static/usage/job_worker.go

```diff
@@ -19,11 +19,13 @@ func CreateUsageJobWorker(
 	cfg config.Config,
 ) func(ctx context.Context) {
 	return func(ctx context.Context) {
-		nodeCount, err := telemetry.CollectNodeCount(ctx, k8sClient)
+		nodelist, err := telemetry.CollectNodeList(ctx, k8sClient)
```

Comment:
calling CollectNodeList and CollectNodeCount doesn't look right here - there is no need for this job to get NodeList, it only needs NodeCount.

Because of that, better to keep CollectNodeList as it was before, or move it here into this package.

Reply:
Does it make sense to create a separate collectNodeCount specific to this job which combines the process? 

The changes make sense in the `collector.go` file as I didn't want to get the nodelist twice (one in CollectNodeCount and the other for the cluster details). So I was thinking of a way that would lead to more code re-use.

Is the tradeoff between this job having more information (the `nodelist`) while being able to re-use the CollectNodeCount from the telemetry package VS the job not storing the `nodelist` but requiring us create a new `collectNodeCount` function worth it? I thought about both and thought the code-reuse would be of better benefit

Reply:
Other collect function returns usage or telemetry related info. However, CollectNodeList is low level - it returns Kubernetes objects. 
It is a better design if they all return types which usage or telemetry-specific.

Better too keep CollectNodeCount as it was.

Because telemetry and usage jobs run every 24 hours, an extra API doesn't make much of a difference.